src/corona_hakab_model/katriel_validation.py

```diff
@@ -0,0 +1,196 @@
+    simulation in days r0 - total number of people that a single individual infects during the whole period of his
+    illness population size p_tau - a vector of length d where d is the number of illness days and p_tau[n] is the
+    probability of infecting others on day n of the illness initial_infected_count_num
+
+    Output:
+    mean - vector of mean number of infected individuals along the num_of_days period
+    sigma - vector of standard deviations of infected individuals along the num_of_days period
+    """
+
+    prev_s = population_size - initial_infected_count_num
+    illness_duration = len(p_tau)
+    prev_i = [0] * (illness_duration - 1) + [initial_infected_count_num]
+    beta = r0 / np.sum(p_tau)  # number of people that a single sick individual infects during a single day (on average)
+
+    mean = np.zeros(num_of_days)  # to store the number of (average) newly infected aside
+    variance = np.zeros(num_of_days)  # to store the std of newly infected aside
+
+    for current_day in range(num_of_days):
+        # calc p of the Binomial distribution:
+        p = 1 - np.exp(-(beta / population_size) * p_tau.dot(prev_i[::-1]))
+        n = prev_s
+        av_newInfections = n * p
+        mean[current_day] = av_newInfections
+        variance[current_day] = n * p * (1 - p)
+        prev_i[0:-1] = prev_i[1:]  # shift old data
+        prev_i[-1] = av_newInfections  # set the new infected as the last entry
+        prev_s = prev_s - av_newInfections
+
+    return mean, np.sqrt(variance)
+
+
+class KatrielConsts(Consts):
+    def __new__(cls, *args, r0, contagiousness, initial_infected_count, num_of_days, illness_duration, **kwargs):
+        self = super().__new__(cls, *args,
+                               r0=r0,
+                               total_steps=num_of_days,
+                               initial_infected_count=initial_infected_count,
+                               **kwargs)
+        self.illness_duration = illness_duration
+        self.contagiousness = contagiousness
+        return self
+
+    def medical_state_machine(self) -> MedicalStateMachine:
+        class SusceptibleTerminalState(SusceptibleState, TerminalState):
+            pass
+
+        class ContagiousStochasticState(ContagiousState, StochasticState):
+            pass
+
+        class ImmuneTerminalState(ImmuneState, TerminalState):
+            pass
+
+        susceptible = SusceptibleTerminalState("Susceptible", test_willingness=0)
+        sick = ContagiousStochasticState(
+            "Sick",
+            contagiousness=self.contagiousness,
+            test_willingness=0,
+        )
+        recovered = ImmuneTerminalState("Recovered", detectable=False, test_willingness=0)
+
+        ret = MedicalStateMachine(susceptible, sick)
+        sick.add_transfer(recovered, dist(self.illness_duration), ...)
+        return ret
+
+
+def run_sim():
+    population_size = 3000
```

Comment:
Shouldn't this be part of config inputed to run_simulation?

Reply:
Nope.
It takes too much time for a bigger population size & it doesn't really work on a smaller population size

Reply:
Can you please document this?